Add backend to Typst

gustavohtc23-sudo opened the pull request:
This pull request adds a new backend to generate tables in Typst format.

The implementation follows the same design and structure used by the existing HTML backend, reusing the same abstractions where possible.

### What was done
- Added a new Typst backend to `PrettyTables.jl`
- Implemented table rendering logic based on the HTML backend approach
- Added tests covering the Typst backend output

### Tests
- New tests were added under the `test/` directory
- The tests validate the generated Typst output against expected results

All tests pass successfully for the new Typst backend.


Closes #209

ronisbr commented on Project.toml:
```diff
@@ -16,6 +16,12 @@ Reexport = "189a3867-3050-52da-a836-e630ba90ab69"
+Typstry = "f0ed7684-a786-439e-b1e3-3b82803b501e"
+
+[extensions]
+PrettyTablesTypstryExt="Typstry"
```
```suggestion
PrettyTablesTypstryExt = "Typstry"
```

ronisbr commented on docs/make.jl:
```diff
@@ -33,13 +33,16 @@ makedocs(
 
 deploydocs(
-    repo = "github.com/ronisbr/PrettyTables.jl.git",
+    repo = "github.com/gustavohtc23-sudo/PrettyTables.jl.git",
```
Please, remove this modification.
```suggestion
    repo = "github.com/ronisbr/PrettyTables.jl.git",
```

ronisbr commented on src/backends/typst/private.jl:
```diff
@@ -0,0 +1,223 @@
+        end
+    end
+end
+function _typst__escape_str(
```
I mean, before the function definition.

ronisbr commented on src/backends/typst/typst_backend.jl:
```diff
@@ -0,0 +1,416 @@
+## Description #############################################################################
+#
+# HTML back end of PrettyTables.jl
```
```suggestion
# Typst back end of PrettyTables.jl
```

ronisbr commented on src/backends/typst/typst_backend.jl:
```diff
@@ -0,0 +1,416 @@
+                # The footnote must be a cell that span the entire printed table.
+                push!(vproperties, "colspan" => string(_number_of_printed_columns(table_data)))
+                merge_style!(vstyle, style.footnote)
+            #     rendered_cell = "#super[$(ps.i)] " * rendered_cell
```
I think this line can be removed right?

ronisbr commented:
Hi @gustavohtc23-sudo !

Awesome work! Thanks!

I have proposed some changes, please verify them. I also want to ask to add `return nothing` statement to those functions that the return is not important (following the blue style). There are also some code commented. If they are not necessary, please, remove them also.

gustavohtc23-sudo commented:
Hi @ronisbr, thank you for the review and the kind words!

I’ve addressed all the suggested changes:
- Added explicit `return nothing` where the return wasn't relevant
- Removed the commented-out code that was no longer necessary.

Please let me know if there is anything else you’d like me to adjust. Thanks again for the feedback!

ronisbr commented:
Hi @gustavohtc23-sudo 

No, the changed I merged helped but there is another problem. Can you please verify?

ronisbr commented:
Now I fixed :) Thanks for the amazing work @gustavohtc23-sudo !
